api/credentialmanager/client.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package credentialmanager
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+
+	"github.com/juju/juju/api/base"
+	"github.com/juju/juju/apiserver/params"
+)
+
+var logger = loggo.GetLogger("juju.api.credentialmanager")
+
+// Client allows access to the credential management API end point.
+type Client struct {
+	base.ClientFacade
+	facade base.FacadeCaller
+}
+
+// NewClient creates a new client for accessing the credential manager API.
+func NewClient(st base.APICallCloser) *Client {
+	frontend, backend := base.NewClientFacade(st, "CredentialManager")
+	return &Client{ClientFacade: frontend, facade: backend}
+}
+
+// InvalidateModelCredential invalidates cloud credential for the model that made a connection.
+func (c *Client) InvalidateModelCredential(reason string) error {
```

Comment:
No, it should not be a bulk call nor a controller facade. It is only ever make sense to be singular and model-scoped.

There will be other functionalities around credentials that will/can be bulk. Some of it will even be at controller level. This her must be model-level. Model credential is currently on the controller but will not be designed like that in the near future.